plugins/modules/cargo.py

```diff
@@ -58,15 +58,87 @@
     type: bool
     default: false
     version_added: 7.5.0
+  force:
+    description:
+      - Forcefully install package(s).
+      - Ignores many checks, and updates/installs forcefully.
+    type: bool
+    default: false
+    required: false
+    version_added: 8.4.0
   state:
     description:
       - The state of the Rust package.
     required: false
     type: str
     default: present
     choices: [ "present", "absent", "latest" ]
+  features:
+    description:
+      - List of features to enable for the package.
+      - Do not use with multiple names.
+    type: list
+    elements: str
+    required: false
+    default: []
+    version_added: 8.4.0
+  all_features:
+    description:
+      - Enable all features for package(s).
+    type: bool
+    default: false
+    required: false
+    version_added: 8.4.0
+  directory:
+    description:
+      - Path to install the package(s) from.
+    type: path
+    required: false
+    version_added: 8.4.0
+  registry:
+    description:
+      - Registry to install the package(s) from.
```

Comment:
I am sorry, but I am afraid I am having trouble understanding what you mean. To clarify: while @radek-sprta is executing `cargo install --list`, I am parsing the `.crates2.json` file located in `CARGO_ROOT`. The options such as `features` or `all_features` come directly from this file, while the registry is extracted from each key within `installs`. Specifically, regarding the `registry` key, there won't be an issue similar to the one described in <https://github.com/ansible-collections/community.general/pull/7363/files#r1367966334>. Regardless of whether it's an update or first installation, you need to specify `registry` each time. If not, the `registry_url` won't align with `metadata["reg"]`, resulting in package reinstallation since it interprets the registry as `default`, exactly as `cargo` does.

Reply:
> Regardless of whether it's an update or first installation, you need to specify `registry` each time. If not, the `registry_url` won't align with `metadata["reg"]`, resulting in package reinstallation since it interprets the registry as `default`, exactly as `cargo` does.

That's fine for cargo to do so, but **not** for an Ansible module (at least not for a new feature). This introduces a breaking change to this module: if you had the package installed with a non-default registry value before, and were running a playbook with the `cargo` module before that specified that this package should be present, that task was not changed before. But with this new feature, the module will suddenly reinstall the package from another registry.

The correct behavior is to only reinstall the package because of registry mismatch if `registry` has been explicitly specified by the user.

(Also I'm not sure whether the format of `.crates2.json` in `CARGO_ROOT` is stable and meant for consumption by third-party code other than `cargo` itself. But that isn't a problem at the moment assuming that the format stayed stable so far.)

Reply:
The current behavior after the latest commit is to not reinstall the package if the state is present, but **will** reinstall the package in case of registry mismatch if the state is latest. Is this the desired behavior?

Reply:
It should only reinstall them if the user explicitly specifies `registry` in the module option and there's a mismatch. If the user does not specify `registry`, it should not reinstall them (at least not because of a registry mismatch).